SingularityService/src/main/java/com/hubspot/singularity/data/SandboxManager.java

```diff
@@ -55,7 +56,7 @@ public SlaveNotFoundException(Exception e) {
       }
 
       return objectMapper.readValue(response.getResponseBodyAsStream(), MESOS_FILE_OBJECTS);
-    } catch (ConnectException ce) {
+    } catch (ExecutionException | ConnectException ce) {
```

Comment:
the issue it that `ConnectException` is the inner exception of `ExecutionException` -- i'd suggest checking the `getCause()` value for `ConnectException`, and throwing SNFE if so and `Throwables.propagate()` if not